### PR TITLE
RFC: Support pre-commit

### PR DESCRIPTION
**Pull request by PaulYuuu**
pre-commit is a framework for managing and maintaining multi-language
pre-commit hooks. It is designed to ensure code quality and consistency
by running checks on code before it is committed to a version control
system, such as Git.

For this PR, add some hooks like isort and black to lint py files.

ID: 2802
refs: https://pre-commit.com
Signed-off-by: Yihuang Yu <yihyu@redhat.com>

### CODE REVIEW

**Comment by PaulYuuu** [comment]
@luckyh FYI

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  - id: check-added-large-files
+    args: ["--maxkb=5120"]
+  - id: check-ast
+  - id: check-yaml
+  - id: end-of-file-fixer
+  - id: mixed-line-ending
+    args: ["--fix=lf"]
+  - id: no-commit-to-branch
+  - id: trailing-whitespace
+- repo: https://github.com/astral-sh/ruff-pre-commit
```
Comment (luckyh) [inline .pre-commit-config.yaml:23]: Let's discuss about if we really want the linters like isort (and other set in the configuration below) to be enforced to this project, **at the current stage**.

@YongxueHong @zhencliu @yanan-fu @vivianQizhu @XueqiangWei any concern?

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -31,3 +31,8 @@ repos:
   hooks:
   - id: pyupgrade
     args: ["--py36-plus"]
+- repo: https://github.com/crate-ci/typos
```
Comment (luckyh) [inline .pre-commit-config.yaml:34]: Suggest separating this commit into a new one.

File: .pre-commit-config.yaml (hunk 1 above)
Comment (vivianQizhu) [reply .pre-commit-config.yaml:23]: Ack from my side.

File: .pre-commit-config.yaml (hunk 2 above)
Comment (PaulYuuu) [reply .pre-commit-config.yaml:34]: Hello @luckyh, do you mean an individual patch? If yes, I will drop it this time and into a new.

Comment (luckyh) [reply .pre-commit-config.yaml:34]: @PaulYuuu yeah, I mean, a new PR. Sorry for the confusion.

Comment (PaulYuuu) [reply .pre-commit-config.yaml:34]: OK, will do that!

**Review by luckyh — APPROVED** [review]
LGTM

**Review by zhencliu — APPROVED** [review]

**Review by vivianQizhu — APPROVED** [review]

**Review by YongxueHong — APPROVED** [review]
LGTM

**Review by XueqiangWei — APPROVED** [review]
LGTM,
Acked-by: xuwei@redhat.com

**Pull request merged**